client/src/components/modals_and_popovers/ModalButton.tsx

```diff
@@ -1,13 +1,26 @@
 import React, { Fragment } from "react";
 
-import { withRouter } from "react-router-dom";
+import { withRouter, RouteComponentProps } from "react-router-dom";
 
 import { SafeJSURL } from "src/general_utils";
 
 import { StatelessModal } from "./StatelessModal";
 
-class ModalButton_ extends React.Component {
-  constructor(props) {
+type ModalButtonProps = RouteComponentProps & {
+  show_condition: { name: string; value: string };
+  match: { params: { options: string } };
```

Comment:
Would expect `match` typing to come from `RouteComponentProps`?

Reply:
> Would expect `match` typing to come from `RouteComponentProps`?

Still curious about this. Is `match: { params: { options: string } };` redundant to `RouteComponentProps`?